otaclient/configs/proxy_info.py

```diff
@@ -0,0 +1,136 @@
+        default=False,
+        # NOTE(20240126): enable_ota_proxy is deprecated,
+        #                 pointing this name to enable_local_ota_proxy.
+        validation_alias=AliasChoices(
+            "enable_local_ota_proxy",
+            "enable_ota_proxy",
+        ),
+    )
+    local_ota_proxy_listen_addr: IPAddressAny = cfg.OTA_PROXY_LISTEN_ADDRESS
+    local_ota_proxy_listen_port: NetworkPort = cfg.OTA_PROXY_LISTEN_PORT
+    # NOTE: this field not presented in v2.5.4,
+    #       for current implementation, it should be default to True.
+    #       This field doesn't take effect if enable_local_ota_proxy is False
+    enable_local_ota_proxy_cache: bool = True
+
+    # NOTE(20240201): check ota_client_log_server_port var in autoware_ecu_setup
+    #                 ansible configurations.
+    LOGGING_SERVER_PORT: NetworkPort = 8083
+    # NOTE: when logging_server is not configured, it implicitly means the logging server
+    #       is located at localhost.
+    logging_server: HTTPURLAny = f"http://127.0.0.1:{LOGGING_SERVER_PORT}"
+
+    def get_proxy_for_local_ota(self) -> str | None:
+        """Tell local otaclient which proxy to use(or not use any)."""
+        if self.enable_local_ota_proxy:
+            # if local otaproxy is enabled, local otaclient also uses it
+            return f"http://{self.local_ota_proxy_listen_addr}:{self.local_ota_proxy_listen_port}"
+        elif self.upper_ota_proxy:
+            # else we directly use the upper proxy
+            return self.upper_ota_proxy
+        # default not using proxy
+
+
+# deprecated field definition
+# <deprecated_old_name> -> <new_field_name>
+_deprecated_field: dict[str, str] = {"enable_ota_proxy": "enable_local_ota_proxy"}
+
+
+def _deprecation_check(_in: dict[str, Any]) -> None:
+    """
+    NOTE: in the future if pydantic support deprecated field annotated, use that
+          mechanism instead of this function.
+    """
+    for old_fname, _ in _in.items():
+        if new_fname := _deprecated_field.get(old_fname):
+            warnings.warn(
+                f"option field '{old_fname}' is superseded by '{new_fname}', "
+                f"and the support for '{old_fname}' option might be dropped in the future. "
+                f"please use '{new_fname}' in proxy_info.yaml instead.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
+
+
+# NOTE: this default is for backward compatible with old device
+#       that doesn't have proxy_info.yaml installed.
+DEFAULT_PROXY_INFO = ProxyInfo(
+    format_version=1,
+    gateway=True,
+    enable_local_ota_proxy=True,
+)
+
+
+def parse_proxy_info(proxy_info_file: StrOrPath) -> ProxyInfo:
+    try:
+        loaded_proxy_info = yaml.safe_load(Path(proxy_info_file).read_text())
+        assert isinstance(loaded_proxy_info, dict), "not a valid yaml file"
+        _deprecation_check(loaded_proxy_info)
+        return ProxyInfo.model_validate(loaded_proxy_info, strict=True)
```

Comment:
[IMO] It would be useful to output the loaded proxy info to the log for analysis.

Reply:
Now a logging containing the content of invalid yaml file will be issued when parsing invalid yaml file.